// @popperjs/core/dist/esm/types.js

--boundary_.oOo._+fHcGNRy1dAuM25iIE7nwEC1A5bBW+kn
Content-Length: 84
Content-Type: application/octet-stream
X-File-MD5: a3a93d198ae13b67ab1d7123a9508d7a
X-File-Mtime: 1706588988
X-File-Path: /Codestack/Level 2/Assignments/Pokemon API Challenge/node_modules/@popperjs/core/dist/esm/utils/math.js

export var max = Math.max;
export var min = Math.min;
export var round = Math.round;
--boundary_.oOo._+fHcGNRy1dAuM25iIE7nwEC1A5bBW+kn
Content-Length: 209
Content-Type: application/octet-stream
X-File-MD5: 7ea00f6aed28f325e8450b13aa13d487
X-File-Mtime: 1706588988
X-File-Path: /Codestack/Level 2/Assignments/Pokemon API Challenge/node_modules/@popperjs/core/dist/esm/utils/getOppositeVariationPlacement.js

var hash = {
  start: 'end',
  end: 'start'
};
export default function getOppositeVariationPlacement(placement) {
  return placement.replace(/start|end/g, function (matched) {
    return hash[matched];
  });
}
--boundary_.oOo._+fHcGNRy1dAuM25iIE7nwEC1A5bBW+kn
Content-Length: 443
Content-Type: application/octet-stream
X-File-MD5: 18220dee44b0cc285a6bd1dd202097ae
X-File-Mtime: 1706588987
X-File-Path: /Codestack/Level 2/Assignments/Pokemon API Challenge/node_modules/@popperjs/core/dist/esm/index.js

export * from "./enums.js";
export * from "./modifiers/index.js"; // eslint-disable-next-line import/no-unused-modules

export { popperGenerator, detectOverflow, createPopper as createPopperBase } from "./createPopper.js"; // eslint-disable-next-line import/no-unused-modules

export { createPopper } from "./popper.js"; // eslint-disable-next-line import/no-unused-modules

export { createPopper as createPopperLite } from "./popper-lite.js";
--boundary_.oOo._+fHcGNRy1dAuM25iIE7nwEC1A5bBW+kn
Content-Length: 184
Content-Type: application/octet-stream
X-File-MD5: cc748e224203097aabaf7380f79f9364
X-File-Mtime: 1706588988
X-File-Path: /Codestack/Level 2/Assignments/Pokemon API Challenge/node_modules/@popperjs/core/dist/esm/utils/mergePaddingObject.js

import getFreshSideObject from "./getFreshSideObject.js";
export default function mergePaddingObject(paddingObject) {
  return Object.assign({}, getFreshSideObject(), paddingObject);
}
--boundary_.oOo._+fHcGNRy1dAuM25iIE7nwEC1A5bBW+kn
Content-Length: 45
Content-Type: application/octet-stream
X-File-MD5: 61c5c51d9f65838107926d86b12dab62
X-File-Mtime: 1706588987
X-File-Path: /Codestack/Level 2/Assignments/Pokemon API Challenge/node_modules/@popperjs/core/dist/umd/enums.min.js.flow

// @flow

export * from '../../lib/enums.js'

--boundary_.oOo._+fHcGNRy1dAuM25iIE7nwEC1A5bBW+kn
Content-Length: 191
Content-Type: application/octet-stream
X-File-MD5: 887af2441fe2b22b9e9566f1256649dc
X-File-Mtime: 1706588988
X-File-Path: /Codestack/Level 2/Assignments/Pokemon API Challenge/node_modules/@popperjs/core/dist/esm/utils/rectToClientRect.js

export default function rectToClientRect(rect) {
  return Object.assign({}, rect, {
    left: rect.x,
    top: rect.y,
    right: rect.x + rect.width,
    bottom: rect.y + rect.height
  });
}
--boundary_.oOo._+fHcGNRy1dAuM25iIE7nwEC1A5bBW+kn
Content-Length: 1335
Content-Type: application/octet-stream
X-File-MD5: 8ae4753699792df63ee836fef2eb5041
X-File-Mtime: 1706588987
X-File-Path: /Codestack/Level 2/Assignments/Pokemon API Challenge/node_modules/@popperjs/core/dist/esm/enums.js

export var top = 'top';
export var bottom = 'bottom';
export var right = 'right';
export var left = 'left';
export var auto = 'auto';
export var basePlacements = [top, bottom, right, left];
export var start = 'start';
export var end = 'end';
export var clippingParents = 'clippingParents';
export var viewport = 'viewport';
export var popper = 'popper';
export var reference = 'reference';
export var variationPlacements = /*#__PURE__*/basePlacements.reduce(function (acc, placement) {
  return acc.concat([placement + "-" + start, placement + "-" + end]);
}, []);
export var placements = /*#__PURE__*/[].concat(basePlacements, [auto]).reduce(function (acc, placement) {
  return acc.concat([placement, placement + "-" + start, placement + "-" + end]);
}, []); // modifiers that need to read the DOM

export var beforeRead = 'beforeRead';
export var read = 'read';
export var afterRead = 'afterRead'; // pure-logic modifiers

export var beforeMain = 'beforeMain';
export var main = 'main';
export var afterMain = 'afterMain'; // modifier with the purpose to write to the DOM (or write into a framework state)

export var beforeWrite = 'beforeWrite';
export var write = 'write';
export var afterWrite = 'afterWrite';
export var modifierPhases = [beforeRead, read, afterRead, beforeMain, main, afterMain, beforeWrite, write, afterWrite];
--boundary_.oOo._+fHcGNRy1dAuM25iIE7nwEC1A5bBW+kn
Content-Length: 252
Content-Type: application/octet-stream
X-File-MD5: 83929ec15ff0fded8c7dd5eb1b3b55c0
X-File-Mtime: 1706588988
X-File-Path: /Codestack/Level 2/Assignments/Pokemon API Challenge/node_modules/@popperjs/core/dist/esm/utils/uniqueBy.js

export default function uniqueBy(arr, fn) {
  var identifiers = new Set();
  return arr.filter(function (item) {
    var identifier = fn(item);

    if (!identifiers.has(identifier)) {
      identifiers.add(identifier);
      return true;
    }
  });
}
--boundary_.oOo._+fHcGNRy1dAuM25iIE7nwEC1A5bBW+kn
Content-Length: 4240
Content-Type: application/octet-stream
X-File-MD5: 18a1ab8dfab101fe6422e7d11982469e
X-File-Mtime: 1706588987
X-File-Path: /Codestack/Level 2/Assignments/Pokemon API Challenge/node_modules/@popperjs/core/dist/umd/enums.js.map

{"version":3,"file":"enums.js","sources":["../../src/enums.js"],"sourcesContent":["// @flow\nexport const top: 'top' = 'top';\nexport const bottom: 'bottom' = 'bottom';\nexport const right: 'right' = 'right';\nexport const left: 'left' = 'left';\nexport const auto: 'auto' = 'auto';\nexport type BasePlacement =\n  | typeof top\n  | typeof bottom\n  | typeof right\n  | typeof left;\nexport const basePlacements: Array<BasePlacement> = [top, bottom, right, left];\n\nexport const start: 'start' = 'start';\nexport const end: 'end' = 'end';\nexport type Variation = typeof start | typeof end;\n\nexport const clippingParents: 'clippingParents' = 'clippingParents';\nexport const viewport: 'viewport' = 'viewport';\nexport type Boundary = Element | Array<Element> | typeof clippingParents;\nexport type RootBoundary = typeof viewport | 'document';\n\nexport const popper: 'popper' = 'popper';\nexport const reference: 'reference' = 'reference';\nexport type Context = typeof popper | typeof reference;\n\nexport type VariationPlacement =\n  | 'top-start'\n  | 'top-end'\n  | 'bottom-start'\n  | 'bottom-end'\n  | 'right-start'\n  | 'right-end'\n  | 'left-start'\n  | 'left-end';\nexport type AutoPlacement = 'auto' | 'auto-start' | 'auto-end';\nexport type ComputedPlacement = VariationPlacement | BasePlacement;\nexport type Placement = AutoPlacement | BasePlacement | VariationPlacement;\n\nexport const variationPlacements: Array<VariationPlacement> = basePlacements.reduce(\n  (acc: Array<VariationPlacement>, placement: BasePlacement) =>\n    acc.concat([(`${placement}-${start}`: any), (`${placement}-${end}`: any)]),\n  []\n);\nexport const placements: Array<Placement> = [...basePlacements, auto].reduce(\n  (\n    acc: Array<Placement>,\n    placement: BasePlacement | typeof auto\n  ): Array<Placement> =>\n    acc.concat([\n      placement,\n      (`${placement}-${start}`: any),\n      (`${placement}-${end}`: any),\n    ]),\n  []\n);\n\n// modifiers that need to read the DOM\nexport const beforeRead: 'beforeRead' = 'beforeRead';\nexport const read: 'read' = 'read';\nexport const afterRead: 'afterRead' = 'afterRead';\n// pure-logic modifiers\nexport const beforeMain: 'beforeMain' = 'beforeMain';\nexport const main: 'main' = 'main';\nexport const afterMain: 'afterMain' = 'afterMain';\n// modifier with the purpose to write to the DOM (or write into a framework state)\nexport const beforeWrite: 'beforeWrite' = 'beforeWrite';\nexport const write: 'write' = 'write';\nexport const afterWrite: 'afterWrite' = 'afterWrite';\nexport const modifierPhases: Array<ModifierPhases> = [\n  beforeRead,\n  read,\n  afterRead,\n  beforeMain,\n
